test-data/unit/check-basic.test

```diff
@@ -372,3 +372,12 @@ none = None
 b = none.__bool__()
 reveal_type(b)  # E: Revealed type is 'builtins.bool'
 [builtins fixtures/bool.pyi]
+
+[case testAssignmentInvariantNote]
+from typing import List
+x: List[int]
+y: List[float]
+y = x # E: Incompatible types in assignment (expression has type "List[int]", variable has type "List[float]") \
+     # N: "List" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance \
+     # N: Consider using "Sequence" instead, which is covariant
+[builtins fixtures/list.pyi]
```

Comment:
Could you please add a similar test for `Dict[str, int]` vs `Dict[str, float]`?

Reply:
Sure!